src/libs/actions/App.js

```diff
@@ -184,6 +185,23 @@ function reconnectApp() {
     });
 }
 
+/**
+ * This promise is used so that deeplink component know when a transition is end.
+ * This is necessary because we want to begin deeplink redirection after the transition is end.
+ */
+let resolveTransitionIsEndPromise;
```

Comment:
I like seeing it here in `App.js` a lot better, yeah. I think that makes more sense. My last suggestion would be to once again rename this (and the other related methods and variables) to be more a little more intuitive.

What do you think about `waitForSignonTransitionToFinish()` and `endSignonTransition()`